pkg/monitor/types.go

```diff
@@ -20,13 +20,27 @@ import (
 	"github.com/aws/aws-node-termination-handler/pkg/node"
 )
 
+const (
+	// SpotITNKind is a const to define a Spot ITN kind of interruption event
+	SpotITNKind = "SPOT_ITN"
+	// ScheduledEventKind is a const to define a scheduled event kind of interruption event
+	ScheduledEventKind = "SCHEDULED_EVENT"
+	// RebalanceRecommendationKind is a const to define a Rebalance Recommendation kind of interruption event
+	RebalanceRecommendationKind = "REBALANCE_RECOMMENDATION"
+	// StateChangeKind is a const to define an EC2 State Change kind of interruption event
+	StateChangeKind = "STATE_CHANGE"
+	// ASGLifecycleKind is a const to define an ASG Lifecycle kind of interruption event
+	ASGLifecycleKind = "ASG_LIFECYCLE"
+)
+
 // DrainTask defines a task to be run when draining a node
 type DrainTask func(InterruptionEvent, node.Node) error
 
 // InterruptionEvent gives more context of the interruption event
 type InterruptionEvent struct {
 	EventID              string
 	Kind                 string
+	Monitor              string
```

Comment:
Let's rename this `AWSEventDetails` (open to suggestions on the name if you disagree). Because the `Kind` field basically contained the monitor type in the past, let's keep that behavior even though the name is sort of wrong. This new field will contain the accurate description of the actual AWS event (e.g., `AWSEventDetails = "SPOT_ITN"` and `Kind = "SQS_TERMINATE"`).

Reply:
Good idea. I like `AWSEventType`.